coreservices/registration/src/CatenaX.NetworkServices.Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs

```diff
@@ -39,17 +40,24 @@ public RegistrationBusinessLogic(IConfiguration configuration, IRegistrationDBAc
 
         public async Task CreateUsersAsync(List<UserCreationInfo> userList, string realm, string token, Dictionary<string, string> userInfo)
         {
-            var manager = new KeycloakIdentityManager(new KeycloakClient(_configuration.GetValue<string>("KeyCloakConnectionString"), () => token), "");
+            var client = new KeycloakClient(_configuration.GetValue<string>("KeyCloakConnectionString"), () => token);
             foreach (UserCreationInfo user in userList)
             {
-                var newUser = new CreateUser
+                var pwd = new Password();
+                var password = pwd.Next();
+                var userToCreate = new User
                 {
                     UserName = user.eMail,
-                    Email = user.eMail,
-                    Groups = new List<string> { user.Role }
+                    Credentials = new List<Credentials>() { new Credentials { Type = "Password", Value = password } },
+                    Enabled = true
                 };
 
-                var password = await manager.CreateUser(realm, newUser);
+                var userId = await client.CreateAndRetrieveUserIdAsync(realm, userToCreate);
+                var clientId = _configuration.GetValue<string>("KeyCloakClientID");
+                var roles = (await client.GetRolesAsync(realm, clientId))
```

Comment:
there is an api-call that allows to directly access a client role by name:
var userId = await GetRoleByNameAsync(realm, string clientId, user.Role).ConfigureAwait(false);

Reply:
Good hint, thanks! I implemented the changes.